### PR TITLE
Support color scheme and syntax for sublime packages

### PR DESCRIPTION
**Pull request by zoli**
- defined [syntax](https://github.com/limetext/lime-backend/blob/12d6e773d04c704f20f28951028b9e5a82b72230/lib/view.go#L61) and [color scheme](https://github.com/limetext/lime-backend/blob/12d6e773d04c704f20f28951028b9e5a82b72230/lib/view.go#L56) interface for registering syntax and color schemes better modularity.

- Any color scheme and syntax should be registered from editor

- Unexported buffer from view #88

### CODE REVIEW

**Comment by zoli** [comment]
Need some tweaks on frontends before merging this.

**Comment by ricochet1k** [comment]
I like the looks of this. Let me know what needs to be done when this is ready and I can update lime-qml.

File: lib/commands/sort.go [hunk 1]
```diff
@@ -73,7 +73,7 @@ func (s textSorter) Less(i, j int) bool {
 
 func (c *SortLinesCommand) Run(v *View, e *Edit) error {
 	sel := v.Sel()
-	buf := v.Buffer()
+	buf := v
```
Comment (erbridge) [inline lib/commands/sort.go]: This no longer makes sense to be called `buf`.

File: lib/commands/sort.go [hunk 2]
```diff
@@ -121,7 +121,7 @@ func (c *SortLinesCommand) Run(v *View, e *Edit) error {
 
 func (c *SortSelectionCommand) Run(v *View, e *Edit) error {
 	sel := v.Sel()
-	buf := v.Buffer()
+	buf := v
```
Comment (erbridge) [inline lib/commands/sort.go]: As above.

**Comment by erbridge** [comment]
This is massive, and hard to review. Do you think you could split out the buffer refactor into a separate PR?

**Comment by zoli** [comment]
[sublime/view_test.go](https://github.com/limetext/lime-backend/blob/81053d888a5f505c3508667c05072e8bc37c4a5c/lib/sublime/view_test.go) contains the tests related to syntax and color schemes for backend view. After merging textmate folder into sublime for avoiding import cycle I had to move those tests from backend/view_test.go to there.

**Pull request merged**